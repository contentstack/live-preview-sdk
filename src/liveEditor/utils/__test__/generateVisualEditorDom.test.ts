import {
    generateVisualEditorCursor,
    generateVisualEditorOverlay,
    generateVisualEditorWrapper,
} from "../generateVisualEditorDom";

describe("generateVisualEditorOverlay", () => {
    const onOverlayClick = jest.fn();

    afterEach(() => {
        onOverlayClick.mockClear();
    });

    test("should generate overlay", () => {
        const overlay = generateVisualEditorOverlay(onOverlayClick);
        expect(overlay).toMatchSnapshot();
    });

    test("should run the function when clicked on the overlay", () => {
        const overlayWrapper = generateVisualEditorOverlay(onOverlayClick);
        const overlay = overlayWrapper.children[0] as HTMLDivElement;

        overlay.click();
        expect(onOverlayClick).toHaveBeenCalledTimes(1);
    });
});

describe("generateVisualEditorCursor", () => {
    test("should generate cursor", () => {
        const cursor = generateVisualEditorCursor();
        expect(cursor).toMatchSnapshot();
    });
});

describe("generateVisualEditorWrapper", () => {
    test("should generate wrapper", () => {
        const overlay = generateVisualEditorOverlay(jest.fn());
        const cursor = generateVisualEditorCursor();

        const wrapper = generateVisualEditorWrapper({
            cursor,
            overlay,
        });
        expect(wrapper).toMatchSnapshot();
    });
});
