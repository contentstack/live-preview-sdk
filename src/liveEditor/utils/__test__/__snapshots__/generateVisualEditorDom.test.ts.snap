// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateVisualEditorCursor should generate cursor 1`] = `
<div
  class="visual-editor__cursor"
  data-testid="visual-editor__cursor"
/>
`;

exports[`generateVisualEditorOverlay should generate overlay 1`] = `
<div
  class="visual-editor__overlay__wrapper"
  data-testid="visual-editor__overlay__wrapper"
>
  
        
  <div
    class="visual-editor__overlay visual-editor__overlay--top"
    data-testid="visual-editor__overlay--top"
  />
  
        
  <div
    class="visual-editor__overlay visual-editor__overlay--left"
    data-testid="visual-editor__overlay--left"
  />
  
        
  <div
    class="visual-editor__overlay visual-editor__overlay--right"
    data-testid="visual-editor__overlay--right"
  />
  
        
  <div
    class="visual-editor__overlay visual-editor__overlay--bottom"
    data-testid="visual-editor__overlay--bottom"
  />
  
        
  <div
    class="visual-editor__overlay--outline"
    data-testid="visual-editor__overlay--outline"
  />
  
    
</div>
`;

exports[`generateVisualEditorWrapper should generate wrapper 1`] = `
<div
  class="visual-editor__container"
  data-testid="visual-editor__container"
>
  <div
    class="visual-editor__cursor"
    data-testid="visual-editor__cursor"
  />
  <div
    class="visual-editor__overlay__wrapper"
    data-testid="visual-editor__overlay__wrapper"
  >
    
        
    <div
      class="visual-editor__overlay visual-editor__overlay--top"
      data-testid="visual-editor__overlay--top"
    />
    
        
    <div
      class="visual-editor__overlay visual-editor__overlay--left"
      data-testid="visual-editor__overlay--left"
    />
    
        
    <div
      class="visual-editor__overlay visual-editor__overlay--right"
      data-testid="visual-editor__overlay--right"
    />
    
        
    <div
      class="visual-editor__overlay visual-editor__overlay--bottom"
      data-testid="visual-editor__overlay--bottom"
    />
    
        
    <div
      class="visual-editor__overlay--outline"
      data-testid="visual-editor__overlay--outline"
    />
    
    
  </div>
</div>
`;
